src/imputors/drop.jl

```diff
@@ -1,73 +1,143 @@
 """
-struct Drop <: Imputor end
+    DropObs(; vardim=2, context=Context)
 
-"""
-    impute!(imp::Drop, ctx::Context, data::AbstractVector)
+Removes missing observations from the `AbstractArray` or `Tables.table` provided.
 
-Uses `filter!` to remove missing elements from the array.
+# Keyword Arguments
+* `vardim=2::Int`: Specify the dimension for variables in matrix input data
+* `context::AbstractContext=Context()`: A context which keeps track of missing data
+  summary information
 
-# Arguments
-* `imp::Drop`: this `Imputor` method
-* `ctx::Context`: contextual information for missing data
-* `data::AbstractVector`: the data to impute
+# Example
+```jldoctest
+julia> using Impute: DropObs, Context, impute
 
-# Returns
-* `AbstractVector`: our data array with missing elements removed
+julia> M = [1.0 2.0 missing missing 5.0; 1.1 2.2 3.3 missing 5.5]
+2×5 Array{Union{Missing, Float64},2}:
+ 1.0  2.0   missing  missing  5.0
+ 1.1  2.2  3.3       missing  5.5
+
+julia> impute(DropObs(; vardim=1, context=Context(; limit=1.0)), M)
+2×3 Array{Union{Missing, Float64},2}:
+ 1.0  2.0  5.0
+ 1.1  2.2  5.5
+```
 """
-function impute!(imp::Drop, ctx::Context, data::AbstractVector)
-    return filter!(x -> !ismissing(ctx, x), data)
+DropObs(; vardim=2, context=Context()) = DropObs(vardim, context)
+
+function impute!(imp::DropObs, data::AbstractVector)
+    imp.context() do c
+        filter!(x -> !ismissing(c, x), data)
+    end
 end
 
-"""
-    impute!(imp::Drop, ctx::Context, data::AbstractMatrix)
+function impute!(imp::DropObs, data::AbstractMatrix)
+    imp.context() do c
+        return filterobs(imp, data) do obs
+            !ismissing(c, obs)
+        end
+    end
+end
 
-Finds the missing rows in the matrix and uses a mask (Vector{Bool}) to return the
-`data` with those rows removed. Unfortunately, the mask approach requires copying the matrix.
+# Deleting elements from subarrays doesn't work so we need to collect that data into
+# a separate array.
+impute!(imp::DropObs, data::SubArray) = impute!(imp::DropObs, collect(data))
 
-NOTES (or premature optimizations):
-* We use `view`, but this will change the type of the `data` by returning a `SubArray`
-* We might be able to do something clever by:
-    1. reshaping the data to a vector
-    2. running `deleteat!` for the appropriate indices and
-    3. reshaping the data back to the desired shape.
+function impute!(imp::DropObs, table)
+    imp.context() do c
+        @assert istable(table)
+        rows = Tables.rows(table)
 
-# Arguments
-* `imp::Drop`: this `Imputor` method
-* `ctx::Context`: contextual information for missing data
-* `data::AbstractMatrix`: the data to impute
+        # Unfortunately, we'll need to construct a new table
+        # since Tables.rows is just an iterator
+        table = Iterators.filter(rows) do r
+            !any(x -> ismissing(c, x), propertyvalues(r))
+        end |> materializer(table)
```

Comment:
I'll note that this goes against the de facto Invenia style

And ask for kindness: please can we just `return materializer(table)` instead?

Reply:
Except it doesn't. At no point does our private style guide comment on the use of `|>`. A bunch of people just don't like that syntax. I think it's useful when you're clearly chaining a set of operations that only take 1 argument.

Reply:
I am not willing to block this PR from being merged over this.

Reply:
yeah, it's "de facto" the style, because it's not used anywhere (becuase a bunch of people dislike it) -- with that in mind it would be a kindness to just move the `materializer` call to the next line 

(but obviously not gonna not-approve over this)